modem: hl7800: ensure net_if_down is triggered and not ignored

rerickson1 opened the pull request:
Trigger network dropped based on socket error notifications.
This debounces the network state and only triggers
the network interface to go down if the network
really drops or causes socket problems.
This will ensure upper networking stack layers can
cleanup broken sockets properly.

Debounce DNS resolver refresh.
Only trigger DNS resolver refresh if the DNS address changes.

bjarki-trackunit commented:
Looks fine :)

rerickson1 commented:
Rebased, no changes compared to before the rebase. @bjarki-trackunit  can you please re-review and signoff?

bjarki-andreasen reviewed (dismissed):
ok

bjarki-trackunit commented:
> Rebased, no changes compared to before the rebase. @bjarki-trackunit can you please re-review and signoff?

Done, I do not have write access though so you still need two more :)

ycsin commented on drivers/modem/hl7800.c:
```diff
@@ -2021,14 +2024,25 @@ static bool on_cmd_atcmdinfo_ipaddr(struct net_buf **buf, uint16_t len)
+	strncpy(temp_addr_str, addr_start, addr_len);
+	temp_addr_str[addr_len] = 0;
 	if (is_ipv4) {
+		ret = strncmp(temp_addr_str, ictx.dns_v4_string,
+							addr_len);
```
I believe this can be fit into one line?

rerickson1 committed:
modem: hl7800: ensure net_if_down is triggered and not ignored

Trigger network dropped based on socket error notifications.
This debounces the network state and only triggers
the network interface to go down if the network
really drops or causes socket problems.
This will ensure upper networking stack layers can
cleanup broken sockets properly.

Debounce DNS resolver refresh.
Only trigger DNS resolver refresh if the DNS address changes.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>
